### ISSUE TITLE
There is no examples on env usage with shell provisioner in documentation

### ISSUE THREAD

**Issue by tuminoid** [mitchellh/vagrant#6707]
I'm trying to use optional env argument with shell provisioner, like this
```
config.vm.provision :shell, :path => "scripts/setup-obs.sh", :env => ENV
```
to pass environment (all of it for test) to he script, but all I get is

```
There are errors in the configuration of this machine. Please fix
the following errors and try again:

shell provisioner:
* The following settings shouldn't exist: env
```

There is no examples of using env in the documentation, and nobody seems to be using it (really hard to google for too).

**Comment by tuminoid** [comment]
Since `env` should behave exactly like `args` as optional argument, might it be that `env` isn't working at all in 1.7.4?

**Comment by sethvargo** [comment]
`env` as not added into Vagrant 1.8. Please upgrade to Vagrant 1.8.

**Comment by tuminoid** [comment]
Well, could it be documented that it is available from 1.8 onwards, and be documented with an example nevertheless?

**Comment by tuminoid** [comment]
For example, documentation says:
```
List of key-value pairs to pass in as environment variables to the script. Vagrant will handle quoting for environment variable values, but the keys remain untouched.
```

If you have `:env => { :foo => 1 }`, you get a stack trace with v1.8.0. It qualifies only `"1"` quoted strings as parameter. `:bar` fails too.

```
/opt/vagrant/embedded/gems/gems/vagrant-1.8.0/plugins/provisioners/shell/provisioner.rb:162:in `quote_and_escape': undefined method `gsub' for 1:Fixnum (NoMethodError)
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.0/plugins/provisioners/shell/provisioner.rb:50:in `block in provision_ssh'
...
```
